docs/graphql/core/api-reference/schema-metadata-api/custom-functions.rst

```diff
@@ -73,6 +75,39 @@ Add an SQL function called ``search_articles`` with a Hasura session argument.
        }
    }
 
+Track ``VOLATILE`` SQL function ``reset_widget`` as a mutation, so it appears
+as a top-level field under the ``mutation`` root field:
+
+.. code-block:: http
+
+   POST /v1/query HTTP/1.1
+   Content-Type: application/json
+   X-Hasura-Role: admin
+
+   {
+       "type": "track_function",
+       "version": 2,
+       "args": {
+           "function": {
+               "schema": "public",
+               "name": "reset_widget"
+           },
+           "configuration": {
+               "exposed_as": "mutation"
+           }
+       }
+   }
+
+If ``exposed_as`` is omitted, the location in the schema to expose the function
+will be inferred from the function's volatility, with ``VOLATILE`` functions
+appearing under the ``mutation`` root, and others ending up under
+``query/subscription``.
+
+.. note::
+
+   It's easy to accidentally give an SQL function the wrong volatility (or for a
+   function to end up with ``VOLATILE`` mistakenly, since it's the default).
```

Comment:
I think this deserves one or two more sentences. Could you add here what would be the scenarios that would make sense (i.e. non-volatile as query, and volatile as mutation), and indicate how the scenario that's odd (volatile as query would be odd, since queries are thought of as non-mutating) and why they are supported regardless (e.g. logging)?

Reply:
Thanks, added a bit of elaboration here.